subsys/net/lib/lwm2m/lwm2m_obj_swmgmt.c

```diff
@@ -0,0 +1,743 @@
+	instance->delete_package_cb = callback_not_defined;
+	instance->activate_cb = callback_not_defined;
+	instance->deactivate_cb = callback_not_defined;
+	instance->write_package_cb = write_callback_not_defined;
+
+#ifdef CONFIG_LWM2M_FIRMWARE_UPDATE_PULL_SUPPORT
+	(void)memset(instance->package_uri, 0, PACKAGE_URI_LEN);
+	instance->pull_context.firmware_ctx.sock_fd = -1;
+	instance->pull_context.result_cb = set_update_result_from_error;
+#endif
+
+	/* initialize instance resource data */
+	INIT_OBJ_RES_DATA(SWMGMT_PACKAGE_NAME_ID, res[index], res_idx, res_inst[index],
+			  res_inst_idx, &instance->package_name, PACKAGE_NAME_LEN);
+
+	INIT_OBJ_RES(SWMGMT_PACKAGE_VERSION_ID, res[index], res_idx, res_inst[index], res_inst_idx,
+		     1, true, &instance->package_version, PACKAGE_VERSION_LEN,
+		     state_read_pkg_version, NULL, NULL, NULL);
+
+	INIT_OBJ_RES_OPT(SWMGMT_PACKAGE_ID, res[index], res_idx, res_inst[index], res_inst_idx, 1,
+			 true, NULL, NULL, package_write_cb, NULL);
+
+#ifdef CONFIG_LWM2M_FIRMWARE_UPDATE_PULL_SUPPORT
+	INIT_OBJ_RES(SWMGMT_PACKAGE_URI_ID, res[index], res_idx, res_inst[index], res_inst_idx, 1,
+		     true, instance->package_uri, PACKAGE_URI_LEN, NULL, NULL, package_uri_write_cb,
+		     NULL);
+#else
+	INIT_OBJ_RES_OPT(SWMGMT_PACKAGE_URI_ID, res[index], res_idx, res_inst[index], res_inst_idx,
+			 1, true, NULL, NULL, package_uri_write_cb, NULL);
+#endif
+
+	INIT_OBJ_RES_EXECUTE(SWMGMT_INSTALL_ID, res[index], res_idx, install_cb);
+
+	INIT_OBJ_RES_OPTDATA(SWMGMT_CHECKPOINT_ID, res[index], res_idx, res_inst[index],
+			     res_inst_idx);
+
+	INIT_OBJ_RES_EXECUTE(SWMGMT_UNINSTALL_ID, res[index], res_idx, uninstall_cb);
+
+	INIT_OBJ_RES_DATA(SWMGMT_UPDATE_STATE_ID, res[index], res_idx, res_inst[index],
+			  res_inst_idx, &instance->update_state, sizeof(uint8_t));
+
+	INIT_OBJ_RES_OPTDATA(SWMGMT_UPDATE_SUPPORTED_OBJECTS_ID, res[index], res_idx,
+			     res_inst[index], res_inst_idx);
+
+	INIT_OBJ_RES_DATA(SWMGMT_UPDATE_RESULT_ID, res[index], res_idx, res_inst[index],
+			  res_inst_idx, &instance->update_result, sizeof(uint8_t));
+
+	INIT_OBJ_RES_EXECUTE(SWMGMT_ACTIVATE_ID, res[index], res_idx, activate_cb);
+	INIT_OBJ_RES_EXECUTE(SWMGMT_DEACTIVATE_ID, res[index], res_idx, deactivate_cb);
+
+	INIT_OBJ_RES_DATA(SWMGMT_ACTIVATION_UPD_STATE_ID, res[index], res_idx, res_inst[index],
+			  res_inst_idx, &instance->activation_state, sizeof(bool));
+
+	INIT_OBJ_RES_OPTDATA(SWMGMT_PACKAGE_SETTINGS_ID, res[index], res_idx, res_inst[index],
+			     res_inst_idx);
+	INIT_OBJ_RES_OPTDATA(SWMGMT_USER_NAME_ID, res[index], res_idx, res_inst[index],
+			     res_inst_idx);
+	INIT_OBJ_RES_OPTDATA(SWMGMT_PASSWORD_ID, res[index], res_idx, res_inst[index],
+			     res_inst_idx);
+
+	inst[index].resources = res[index];
+	inst[index].resource_count = res_idx;
+	return &inst[index];
+}
+
+static int lwm2m_swmgmt_init(const struct device *dev)
+{
+	swmgmt.obj_id = LWM2M_OBJECT_SOFTWARE_MANAGEMENT_ID;
+	swmgmt.fields = fields;
```

Comment:
You should also set version_major/minor fields.

Reply:
Setting it to 1.1

Reply:
Why 1.1? I see in [OMA registry](https://technical.openmobilealliance.org/OMNA/LwM2M/LwM2MRegistry.html) that this object is only available in version 1.0.